src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/MonitorAllTripsJob.java

```diff
@@ -46,22 +47,50 @@ public void run() {
         }
 
         try {
-            // request all monitored trips from the mongo collection
-            for (MonitoredTrip monitoredTrip : Persistence.monitoredTrips.getAll()) {
-                // attempt to add trip to tripAnalysisQueue until a spot opens up in the queue. If the timeout is
+            // Request all monitored trip IDs from the Mongo collection, all at once, and loop through a list of strings.
+            // If we looped using a Mongo-provided iterator instead, and the Mongo connection is dropped for any reason
+            // while the iterator is open, this thread would become blocked and
+            // prevent subsequent trip monitoring jobs to start.
+            // Performance note: Don't retrieve the full data for each trip at this time.
+            // This saves bandwidth and memory, as we don't use the trip data immediately besides the ID.
+            // The full data for each trip will be fetched at the time the actual analysis takes place.
+            // TODO: Filter out trips that would be skipped by the CheckMonitoredTrip.
```

Comment:
Can this TODO be done now? If not, perhaps explain why. TODOs seem to have a habit of not getting done.